### ISSUE TITLE
Question about choice of Discriminator

### ISSUE THREAD

**Issue by zhenbuxianggaimingzi** [RedMediaTech/ODTSR#5]
Great work! I was wondering why do you use wan2.1 as the discriminator? It seems that the paper did not mention the reasons for the choice or the related ablation experiments

**Comment by Feynman1999** [comment]
In general, we prefer the generator and the discriminator to operate in a consistent latent space. Conveniently, Wan2.1 and Qwen-Image share the same encoder, so we directly adopt Wan2.1 as the discriminator. Although Wan2.1 is a temporal DiT model with a three-dimensional (T–H–W) structure, it still works well when T=1.

In addition, the 1.3B-parameter version of Wan2.1 is relatively lightweight and offers good training speed.

**Comment by zhenbuxianggaimingzi** [comment]
Got it ! Thank you for the explanation! 
 
Additionally,  I would like to ask, have you ever tried decoding latent to the rgb space for GAN training? Is there any difference between this and directly training GAN in the latent space? or do these two have their own advantages and disadvantages?

**Comment by Feynman1999** [comment]
We have tried, for example, returning to RGB and using dinov2/dinov3 for adversarial training, and the results were also quite good.
(Actually, dinov3's processing of RGB directly performs a conv16x16 downsampling, which is essentially also a kind of "latent space".)
My focus points on the discriminator are: 1. How is the training efficiency? 2. What is the receptive field of the features/latent learned by the discriminator? 3. Using loss curves to diagnose the training situation.

ref:

<img width="598" height="567" alt="Image" src="https://github.com/user-attachments/assets/0081f43d-bc31-47cf-b8a5-df69266d4129" />

**Comment by zhenbuxianggaimingzi** [comment]
Yes, these three points are very important criteria~

In fact, I have also conducted a similar experiment, using DINOv2 as the discriminator to train the GAN in the RGB space. Judging from the loss curve, everything was relatively normal. However, the clarity of the final test set was not very ideal. From this perspective, it might be the additional Noise-hybrid Visual Stream module you have here. It has a greater effect than the module I designed myself, rather than the difference between training GAN in the latent space or the rgb space

Thank you again for your explanation. It's indeed a very good and inspiring job~

**Comment by Feynman1999** [comment]
> Yes, these three points are very important criteria~
> 
> In fact, I have also conducted a similar experiment, using DINOv2 as the discriminator to train the GAN in the RGB space. Judging from the loss curve, everything was relatively normal. However, the clarity of the final test set was not very ideal. From this perspective, it might be the additional Noise-hybrid Visual Stream module you have here. It has a greater effect than the module I designed myself, rather than the difference between training GAN in the latent space or the rgb space
> 
> Thank you again for your explanation. It's indeed a very good and inspiring job~

thanks~
Based on your description of your method, I can think of possible reasons why your approach might not work:
1. Check out the R3GAN concept (https://github.com/brownvc/R3GAN) — we found this to work better
2. The Noise-hybrid Visual Stream preserves the base model to the greatest extent possible (the double stream is completely untrained), and denoising the noisy low-quality images aligns with the original pretrained model.

**Comment by zhenbuxianggaimingzi** [comment]
Thank you for your suggestion! I'll check out R3GAN and do some experiments ~
